### PR TITLE
[# 154378274] As an investor, so I know the attribution requirements for any open source licenses we use are met, create an s3 bucket and folder arrangement where license reports can be placed when we release software.

### PR DESCRIPTION
**Pull request by ro31337**
As an investor, so I know the attribution requirements for any open source licenses we use are met, create an s3 bucket and folder arrangement where license reports can be placed when we release software.

[Tracker #154378274](https://www.pivotaltracker.com/story/show/154378274)

### CODE REVIEW

**Comment by ro31337** [comment]
@cobbr2 can you take a look?

Here is how to test:

1. Update `default_projects` so it won't take forever to test. For example:

```
default_projects="
  /home/rpushkin/work/stone
  /home/rpushkin/work/banyan
  "
```

2. Run:

```
$ aws-environment uat
$ ./bin/gr_licenses --html --s3
```

Also, I reverted 4 commits and left 2 minor commits (bugfixes) in your branch.

File: bin/gr_licenses [hunk 1]
```diff
@@ -62,15 +83,29 @@ while [ $# -gt 0 ] ; do
...
-    sed -e '/^<csv.*>/d' -e '/^LicenseFinder::/d' |\
-    sort >"${basename}.csv"
+function make_csv_or_html() {
+  # Remove everything before and after result tags,
+  # Remove result tags and debug info,
+  # Sort if format is csv
+  sed -n -e '/<result>/,/<\/result>/p' "$1.txt" |\
+    sed -e '/^<result>/d' -e '/^<\/result>/d' -e '/^LicenseFinder::/d' |\
+    (test $format == "csv" && sort || cat) > "$1.$format"
+}
```
Comment (cobbr2) [inline bin/gr_licenses:99]: Test question: Does this successfully filter out the lines that identify which package managers are being used?

File: bin/gr_licenses [hunk 2]
```diff
@@ -62,15 +83,29 @@ while [ $# -gt 0 ] ; do
...
+
+function s3_upload() {
+  if [ -n "${aws_environment}" ] ; then
+    local path="$1.$format"
+    local filename="$2.$format"
+    aws s3 rm s3://grnds-${aws_environment}-licenses/${filename} > /dev/null 2>&1
+    aws s3 cp ${path} s3://grnds-${aws_environment}-licenses --acl public-read && \
```
Comment (cobbr2) [inline bin/gr_licenses]: Do we need to do the `rm` before the `cp`? I would prefer not to have a blank page in the case of an error.

**Review by cobbr2 — APPROVED** [review]
It's moving in the right direction. I think we have two other requirements that I wasn't clear on in the original story writing.  Also, this work still doesn't match the title; that title was just about the chore of deciding how these were going to look and creating the bucket.  I'll rename again... but when we change plans, we should change the written plan, too.

**Comment by cobbr2** [comment]
Tested fine using banyan (`bin/gr_licenses --html banyan --s3`)

**Pull request merged**